lib/Runtime/Library/WebAssemblyModule.cpp

```diff
@@ -281,9 +282,14 @@ WebAssemblyModule::ValidateModule(
     {
         Js::AutoDynamicCodeReference dynamicFunctionReference(scriptContext);
         SRCINFO const * srcInfo = scriptContext->cache->noContextGlobalSourceInfo;
-        Js::Utf8SourceInfo* utf8SourceInfo = Utf8SourceInfo::New(scriptContext, (LPCUTF8)buffer, lengthBytes / sizeof(char16), lengthBytes, srcInfo, false);
 
-        Wasm::WasmModuleGenerator bytecodeGen(scriptContext, utf8SourceInfo, (byte*)buffer, lengthBytes);
+        // review: unsure if we need copy here, but seems safer to do it
```

Comment:
Since the created module doesn't escape this scope, it should be garbage collected asap and there should be no further references to the buffer after this call. Therefore, we don't need to get a copy of the buffer, simply validate it then trash all references to it.